main.go

```diff
@@ -278,6 +300,7 @@ func sendSamples(logger log.Logger, w writer, timeseries []*prompb.TimeSeries) {
 	if err != nil {
 		level.Warn(logger).Log("msg", "Error sending samples to remote storage", "err", err, "storage", w.Name(), "num_samples", len(timeseries))
 		failedSamples.WithLabelValues(w.Name()).Add(float64(len(timeseries)))
+		writeErrors.Inc()
 	}
 	sentSamples.WithLabelValues(w.Name()).Add(float64(len(timeseries)))
```

Comment:
I know this is from previous code, but can you move this line into an `else` statement?
We shouldn't count the sample if it wasn't successful